integration_tests/src/main/python/cast_test.py

```diff
@@ -863,3 +863,18 @@ def test_cast_date_integral_and_fp_ansi_off():
         lambda spark: unary_op_df(spark, date_gen).selectExpr(
             "cast(a as boolean)", "cast(a as byte)", "cast(a as short)", "cast(a as int)", "cast(a as long)", "cast(a as float)", "cast(a as double)"),
             conf=ansi_disabled_conf)
+
+
+# ToPrettyString is generated by df.show(), and there is no ToPrettyString function in pyspark.sql.functions package
+# Here use df.show() to test ToPrettyString on date column. If fall back occurs, `with_gpu_session` will report error:
+#   Part of the plan is not columnar class
+@allow_non_gpu('CollectLimitExec')
```

Comment:
NIT: we do not need to run this test before ToPrettyString is supported. That is before Spark 350.

Reply:
I would sugget moving the tests to Scala world where we can check the existence of `GpuToPrettyString` easier.
e.g.
```
XXTest extends SparkQueryCompareTestSuite {

  test("ToPrettyString always on GPU") {
    val  (_, gpuPlan) = executeFunOnCpuAndGpuWithCapture(
      spark => spark.createDataFrame(xxxx),
      df => df.show
      conf = new SparkConf().set('spark.rapids.sql.hasExtendedYearValues': False)
                                              .set('spark.sql.session.timeZone': 'America/Los_Angeles')
    )
    ExecutionPlanCaptureCallback.assertContains(gpuPlan, "GpuToPrettyString")
 }
}
```

Reply:
Could we use Python test case since this comment is a NIT. @firestarman do you insist on using Scala test case?